### ISSUE TITLE
Expose the "Spelling and Grammar" related APIs 

### ISSUE THREAD

**Issue by zcbenz** [electron/electron#3211]
On OS X apart from the basic spell-checking it also provides a "Spelling and Grammar" menu that allows some more advanced operations:
<img width="666" alt="screen shot 2015-10-26 at 10 02 39 am" src="https://cloud.githubusercontent.com/assets/639601/10719716/bff72fd4-7bc8-11e5-9b97-980e9a75dd91.png">

Chrome browser has implemented most of the features, we should find a way to make Electron apps use it without any pain.

**Comment by rodrigok** [comment]
Any news here?

We are trying to use `node-spellchecker` in latest electron release at Rocket.Chat with no lucky

**Comment by danhp** [comment]
I'd recommend having a look at https://github.com/paulcbetts/electron-spellchecker

**Comment by rodrigok** [comment]
@danhp I tried too, not even the example works.

I'm using node 4.6, Idk if it's recommended.

**Comment by codebytere** [comment]
There are third-party implementations available for now, and as this hasn't been active in quite some time and isn't within our immediate bandwidth i'm going to go ahead and close it.

**Comment by akashnimare** [comment]
@codebytere most of those **third-party** implementations are broken. They either don't work at all or work partially with tons of bug. This one feature is must have in most of the electron apps. I'm not sure closing this issue is a right decision 😞

**Comment by codebytere** [comment]
@akashnimare ok! if they're all broken we can definitely revisit what might be the best path forward for this :) i'll re-open in the meantime.

**Comment by akashnimare** [comment]
@codebytere thanks :) a built-in support for the spellchecker will make electron more awesome ❤️

**Comment by ccorcos** [comment]
I agree -- I'm using [electron-spellchecker](https://github.com/electron-userland/electron-spellchecker) but the "Add to Dictionary" doesn't work and there are some [performance issues](https://github.com/electron-userland/electron-spellchecker/issues/95) with spell check running synchronously every time you press a character. Exposing APIs baked into electron would be amazing! ❤️

**Comment by zeke** [comment]
I seem to recall the folks at Slack having expertise in this area. Maybe @mgc or @felixrieseberg can shed some light on the state of spellchecking in userland?

**Comment by akashnimare** [comment]
@zeke also cc-ing @paulcbetts :)

**Comment by nitsakh** [comment]
The spellchecker API has changed and is no longer synchronous!

**Comment by WhiteNoiseDev** [comment]
@zcbenz I saw you did some work in this area. Is there a way to programmatically enable/disable "check Grammar with Spelling"?

**Comment by OogieM** [comment]
Also please allow electron apps to use both spellcheck and grammer as other Apple apps do.
